src/Language/PureScript/Errors.hs

```diff
@@ -599,18 +599,17 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
             , line "with type"
             , markCodeBox $ indent $ typeAsBox t2
             ]
-    renderSimpleErrorMessage (OverlappingInstances nm ts (d : ds)) =
+    renderSimpleErrorMessage (OverlappingInstances _ _ []) = internalError "OverlappingInstances: empty instance list"
+    renderSimpleErrorMessage (OverlappingInstances nm ts ds) =
       paras [ line "Overlapping type class instances found for"
             , markCodeBox $ indent $ Box.hsep 1 Box.left
                 [ line (showQualified runProperName nm)
                 , Box.vcat Box.left (map typeAtomAsBox ts)
                 ]
             , line "The following instances were found:"
-            , indent $ paras (line (showQualified showIdent d <> " (chosen)") : map (line . showQualified showIdent) ds)
-            , line "Overlapping type class instances can lead to different behavior based on the order of module imports, and for that reason are not recommended."
-            , line "They may be disallowed completely in a future version of the compiler."
+            , indent $ paras (map (line . showQualified showIdent) ds)
+            , line "Overlapping type class instances can lead to different behavior based on the order of module imports, and for that reason are disallowed."
```

Comment:
I'm wondering if there is any point in mentioning the order of module imports bit. Given that they are supposed to be disallowed anyway, you're not supposed to be able to observe the described "different behavior" because you'd just see this error message. (Although the overlap check does need improving so that it catches all scenarios).

Reply:
I'm also tempted to get rid of this line. We can give a more detailed explanation on the wiki.

Reply:
Thanks guys for comments!

I removed the line with the hint why overlapping instances are bad and added myself to contributors.

> (Although the overlap check does need improving so that it catches all scenarios).

If there is something else should be done outside of the file it's probably easier to close a PR. Since I need pointers of how to do that.